src/Layer/Potree2Layer.js

```diff
@@ -0,0 +1,98 @@
+import * as THREE from 'three';
+import PointCloudLayer from 'Layer/PointCloudLayer';
+import Potree2Node from 'Core/Potree2Node';
+import Extent from 'Core/Geographic/Extent';
+import Potree2Utils from 'Utils/Potree2Utils';
+
+const bboxMesh = new THREE.Mesh();
+const box3 = new THREE.Box3();
+bboxMesh.geometry.boundingBox = box3;
+
+/**
+ * @property {boolean} isPotreeLayer - Used to checkout whether this layer
+ * is a Potre2eLayer. Default is `true`. You should not change this, as it is
+ * used internally for optimisation.
+ */
+class Potree2Layer extends PointCloudLayer {
+    /**
+     * Constructs a new instance of Potree2 layer.
+     *
+     * @constructor
+     * @extends PointCloudLayer
+     *
+     * @example
+     * // Create a new point cloud layer
+     * const points = new Potree2Layer('points',
+     *  {
+     *      source: new Potree2Source({
+     *          url: 'https://pointsClouds/',
+     *          file: 'metadata.json',
+     *      }
+     *  });
+     *
+     * View.prototype.addLayer.call(view, points);
+     *
+     * @param {string} id - The id of the layer, that should be unique. It is
+     * not mandatory, but an error will be emitted if this layer is added a
+     * {@link View} that already has a layer going by that id.
+     * @param {Object} config - Configuration, all elements in it
+     * will be merged as is in the layer. For example, if the configuration
+     * contains three elements `name, protocol, extent`, these elements will be
+     * available using `layer.name` or something else depending on the property
+     * name. See the list of properties to know which one can be specified.
+     * @param {string} [config.crs=ESPG:4326] - The CRS of the {@link View} this
+     * layer will be attached to. This is used to determine the extent of this
+     * layer.  Default to `EPSG:4326`.
+     */
+    constructor(id, config) {
+        super(id, config);
+        this.isPotreeLayer = true;
+
+        const resolve = this.addInitializationStep();
+
+        this.source.whenReady.then((metadata) => {
+            this.scale = new THREE.Vector3(1, 1, 1);
```

Comment:
Why are you not using `metadata.scale` ?
```suggestion
            this.scale = new THREE.Vector3().fromArray(metadata.offset);
```

Reply:
Because scale is already applied in decoder, it will be apply twice.